### ISSUE TITLE
sonobuoy-e2e-job ImagePullBackOff in air-gapped.

### ISSUE THREAD

**Issue by senpot** [vmware-tanzu/sonobuoy#1376]
**What steps did you take and what happened:**
I executed following command, but sonobuoy-e2e-job pod status is ImagePullBackOff.

```bash
$ sonobuoy run \
  --sonobuoy-image myrepo.com/${PROJECT_NAME}-sonobuoy:v0.53.0 \
  --e2e-repo-config conformance-image-config.yaml \
  --kube-conformance-image myrepo.com/conformance:v1.19.6 \
  --systemd-logs-image myrepo.com/sonobuoy/systemd-logs:v0.3 \
  --kubernetes-version v1.19.6 \
  --level debug \
  --wait
INFO[0000] created object                                name=sonobuoy namespace= resource=namespaces
INFO[0000] created object                                name=sonobuoy-serviceaccount namespace=sonobuoy resource=serviceaccounts
INFO[0000] created object                                name=sonobuoy-serviceaccount-sonobuoy namespace= resource=clusterrolebindings
INFO[0000] created object                                name=sonobuoy-serviceaccount-sonobuoy namespace= resource=clusterroles
INFO[0000] created object                                name=sonobuoy-config-cm namespace=sonobuoy resource=configmaps
INFO[0000] created object                                name=sonobuoy-plugins-cm namespace=sonobuoy resource=configmaps
INFO[0001] created object                                name=sonobuoy namespace=sonobuoy resource=pods
INFO[0001] created object                                name=plugin-e2e-cm namespace=sonobuoy resource=configmaps
INFO[0001] created object                                name=sonobuoy-aggregator namespace=sonobuoy resource=services
```

```bash
# kubectl describe pod sonobuoy-e2e-job-c030bd3fd2414d87 -n sonobuoy
Name:         sonobuoy-e2e-job-c030bd3fd2414d87
Namespace:    sonobuoy
(snip....)
Containers:
  e2e:
    Container ID:
    Image:         k8s.gcr.io/conformance:v1.19.6
    Image ID:

(snip....)

Events:
  Type     Reason     Age                From               Message
  ----     ------     ----               ----               -------
  Normal   heduled  30s                default-scheduler  Successfully assigned sonobuoy/sonobuoy-e2e-job-c030bd3fd2414d87 to ip-10-0-44-185.ap-northeast-1.compute.internal
  Normal   Pulling    29s                kubelet            Pulling image "k8s.gcr.io/conformance:v1.19.6"
  Warning  Failed     14s                kubelet            Failed to pull image "k8s.gcr.io/conformance:v1.19.6": rpc error: code = Unknown desc = Error response from daemon: Get https://k8s.gcr.io/v2/: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)
  Warning  Failed     14s                kubelet            Error: ErrImagePull
  Normal   BackOff    11s (x2 over 12s)  kubelet            Back-off pulling image "k8s.gcr.io/conformance:v1.19.6"
  Warning  Failed     11s (x2 over 12s)  kubelet            Error: ImagePullBackOff
```

**What did you expect to happen:**

I think this option `--kube-conformance-image` support pulling image (conformance) in air-gapped. 

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

conformance-image-config.yaml
```yaml
buildImageRegistry: myrepo.com
dockerGluster: myrepo.com
dockerLibraryRegistry: myrepo.com
e2eRegistry: myrepo.com
e2eVolumeRegistry: myrepo.com
gcRegistry: myrepo.com
gcEtcdRegistry: myrepo.com
promoterE2eRegistry: myrepo.com
sigStorageRegistry: myrepo.com
```

I can pull image myrepo.com/conformance:v1.19.6.

**Environment:**

- Sonobuoy version: 
  - v0.53.0
- Kubernetes version: (use `kubectl version`):
```bash
Client Version: version.Info{Major:"1", Minor:"21", GitVersion:"v1.21.1", GitCommit:"5e58841cce77d4bc13713ad2b91fa0d961e69192", GitTreeState:"clean", BuildDate:"2021-05-12T14:18:45Z", GoVersion:"go1.16.4", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"19+", GitVersion:"v1.19.13-eks-8df270", GitCommit:"8df2700a72a2598fa3a67c05126fa158fd839620", GitTreeState:"clean", BuildDate:"2021-07-31T01:36:57Z", GoVersion:"go1.15.14", Compiler:"gc", Platform:"linux/amd64"}
```
- Kubernetes installer & version:
  - EKS(v1.19) on AWS
- Cloud provider or hardware configuration:
- OS (e.g. from `/etc/os-release`): 
- Sonobuoy tarball (which contains * below)
  - sonobuoy_0.53.0_linux_amd64.tar.gz

**Comment by johnSchnake** [comment]
👍 Thanks for the report, should be easy to repro and diagnose. Will get a patch in for this quickly I'm sure.

**Comment by johnSchnake** [comment]
Confirmed that the flag modifies the _plugins_ image but with the last change to the flag does not properly change the `workerImage` field of the config. That would lead to the aggregator being launched correctly but the plugin workers failing to pull, which is what we see here.

**Comment by johnSchnake** [comment]
OK, I realize what is happening after I copied your given command.

You are specifying both the kube-conformance-image and the kubernetes-version flags and they are effectively fighting.

There was some finagling of the flags recently and we even added a little warning if you specified the `--kube-conformance-image` and `kube-conformance-image-version` flags to let you know that (1) you shouldn't and (2) we don't promise a clear resolution. We also made `kube-conformance-image-version` deprecated and you _should_ use `kubernetes-version` but we failed to include the warning if that was the case.

So, mitigation today: just run this command without the kubernetes-version flag. It should run with your given conformance image just fine.

TBD: Should sonobuoy allow you to specify these together at all? I'll include a fix so you'd get the same warning I expected but will leave this open for a bit more until I decide the full resolution. Let me know if you have any thoughts on the topic.
